src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,116 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import duke.exceptions.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * The Storage class handles loading and saving tasks to a file.
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path to the file where tasks are stored.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads tasks from the file and returns them as an ArrayList.
+     *
+     * @return An ArrayList containing the loaded tasks.
+     * @throws DukeException If there is an error while loading tasks.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> dukeList = new ArrayList<>();
+        String[] splitFilePath = this.filePath.split("/");
+        String foldersPath = String.join("/", Arrays.copyOf(splitFilePath, splitFilePath.length - 1));
+        new File(foldersPath).mkdirs();
+        boolean fileExists;
```

Comment:
I like how you name your variable name. They are all in camelCase and easy to understand. The boolean fileExists also sound like a boolean.